modules/rosetta/src/test/scala/org/tessellation/rosetta/domain/construction/ConstructionServiceSuite.scala

```diff
@@ -149,11 +149,73 @@ object ConstructionServiceSuite extends MutableIOSuite with Checkers with Transa
         Transfer,
         none,
         AccountIdentifier(address, none).some,
-        Amount(AmountValue(1L), Currency(CurrencySymbol("DAG"), CurrencyDecimal(8L))).some
+        Amount(AmountValue(1L), currency.Currency(currency.CurrencySymbol("DAG"), currency.CurrencyDecimal(8L))).some
       )
 
       val accountIdentifiers = cs.getAccountIdentifiers(List(operation))
       expect.eql(none, accountIdentifiers)
     }
   }
+
+  test("returns operations for a valid signed transaction hex") { res =>
```

Comment:
there is no test checking whether the negative or positive operation is assigned to the correct address.
@kpudlik just found out that you use destination address twice

Reply:
Exactly. It returns operations but operations are incorrect due to a mistake in a code.

You should check following things:

1. `result.operations.lenght == 2`
2. `result.operations[0].address == srcAddress && result.operations[1].address == dstAddress`
3. `result.operations[0].amount = -result.operations[1].amount`

Reply:
Added these tests.